FrameworkInternals/DesignInspector.py

```diff
@@ -114,9 +114,7 @@ def class_has_legit_device_parent(self, class_name):
             return False # referenced from multiple parents, so far sure no unique parent
         if has_objects_origins[0] == "Root":
             return True # Root by definition has device logic
-        if not self.class_has_device_logic(has_objects_origins[0]):
-            return False
-        return True
+        return self.class_has_device_logic(has_objects_origins[0])
```

Comment:
Let’s keep the explicit return True and return False statements for clarity and to ensure consistent boolean results, which makes the code more maintainable and robust against future changes.

Reply:
OK, reverted DesignInspector, now as master

Inverted if/else logic in template

Should be good to go, have a look and see what you think